tools/node-modules

```diff
@@ -56,35 +56,96 @@ EOF
     clone_from_cache "${sha}"
 }
 
-cmd_rebuild() {
-    for file in npm-shrinkwrap.json package-lock.json yarn.lock; do
-        if [ -e "${file}" ]; then
-            echo "Please delete ${file} from $(pwd) and start again"
-            exit 1
-        fi
-    done
+cmd_npm_install() {
+    if [ -f /run/.toolboxenv ]; then
+        exec flatpak-spawn --host -- "$0" npm_install "$@"
+        exit 1
+    fi
+
+    CACHE="cockpit-project-npm-cache-volume"
+    IMAGE="docker.io/library/node:alpine"
+
+    # make sure the node user can write to the cache volume
+    podman run \
+        --rm \
+        --volume "${CACHE}":/home/node/.npm:U \
+        "${IMAGE}" chown -R node:node /home/node >&2
+
+    # do the actual work
+    podman run \
+        --log-driver='none' \
+        --rm \
+        --user node \
+        --workdir /home/node \
+        --interactive \
+        --attach stdin \
+        --attach stdout \
+        --attach stderr \
+        --volume "${CACHE}":/home/node/.npm \
+        "${IMAGE}" /bin/sh -c '
+            set -eux
+            tee package.json >/dev/null
```

Comment:
I had to stare at this for a bit. `cat > package.json` would be a bit clearer to me.

Reply:
This is what I had at first.  This is about the `set -x` and the output we see.  `tee package.json` is more instructive than `cat` (which is what you see with your suggestion).  I had also tried `dd of=package.json` but then I got the annoying transfer statistics line... and turning it off would have made it even more ugly...

Other suggestions?

Reply:
Ack, seeing the file name in `set -x` makes sense, thanks!